### PR TITLE
Adding default build flags

### LINKED ISSUE #2033: some labels should be added to every container automatically at build time
## Version of Singularity:

master

### Expected behavior

On a newly built container one should be able to do this

```
$ singularity inspect my_container.sif
{
    "org.label-schema.usage.singularity.deffile.bootstrap": "docker",
    "org.label-schema.usage.singularity.deffile": "my_container.def",
    "org.label-schema.usage": "/.singularity.d/runscript.help",
    "org.label-schema.schema-version": "1.0",
    "org.label-schema.usage.singularity.deffile.from": "busybox",
    "org.label-schema.build-date": "Tue,_11_Sep_2018_10:43:13_-0400",
    "org.label-schema.usage.singularity.runscript.help": "/.singularity.d/runscript.help",
    "org.label-schema.usage.singularity.version": "2.6.0-HEAD.579c415",
    "org.label-schema.build-size": "2MB"
}
```

### Actual behavior

With a newly built container in 3.0 one gets this:

```
$ singularity inspect my_container.sif 
/bin/cat: .singularity.d/labels.json: No such file or directory
```

### PR DESCRIPTION
**Pull request by ikaneshiro**
Adds things like boostrap header information, build time and singularity version used to build
fixes #2033

```singularity shell test.sif
Singularity> cat /.singularity.d/labels.json 
{
	"org.label-schema.build-date": "Friday_21_September_2018_10:10:56_EDT",
	"org.label-schema.schema-version": "1.0",
	"org.label-schema.usage": "/.singularity.d/runscript.help",
	"org.label-schema.usage.singularity.deffile.bootstrap": "docker",
	"org.label-schema.usage.singularity.deffile.from": "alpine",
	"org.label-schema.usage.singularity.runscript.help": "/.singularity.d/runscript.help",
	"org.label-schema.usage.singularity.version": "v3.0.0-alpha.2-44-gd6ec242b"
}

### CODE REVIEW

File: src/pkg/build/assemblers/assembler_sandbox.go [hunk 1]
```diff
@@ -126,60 +128,98 @@ func insertDefinition(b *types.Bundle) error {
 	return nil
 }
 
-func insertLabelsJSON(b *types.Bundle) error {
+func insertLabelsJSON(b *types.Bundle) (err error) {
+
```
Comment (ArangoGutierrez) [inline src/pkg/build/assemblers/assembler_sandbox.go]: white line

File: src/pkg/build/assemblers/assembler_sandbox.go [hunk 2]
```diff
@@ -126,60 +128,98 @@ func insertDefinition(b *types.Bundle) error {
...
+		err = json.Unmarshal(jsonBytes, &labels)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
+
+func addBuildLabels(labels map[string]string, b *types.Bundle) error {
+
```
Comment (ArangoGutierrez) [inline src/pkg/build/assemblers/assembler_sandbox.go]: white line

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

**Review by GodloveD — CHANGES_REQUESTED** [review]
These changes do add labels to a container when built.  But now the `%labels` section doesn't seem to work anymore.

**Comment by ikaneshiro** [comment]
@GodloveD That should be fixed now

**Comment by bauerm97** [comment]
@GodloveD can you please check if bug is fixed, also would be great if we could add a test to catch this regression ;) @ikaneshiro

**Comment by GodloveD** [comment]
I was just looking at it.  Seems to behave as advertised now.

**Review by GodloveD — APPROVED** [review]

**Review by ArangoGutierrez — APPROVED** [review]

**Pull request merged**